src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -1340,6 +1348,20 @@ private void AddInputKeysLookupLocked(TransactionData transactionData)
             }
         }
 
+        /// <summary>
+        /// Remove from the list of unspent outputs kept in memory.
+        /// </summary>
+        private void RemoveInputKeysLookupLock(TransactionData transactionData)
+        {
+            Guard.NotNull(transactionData, nameof(transactionData));
+            Guard.NotNull(transactionData.SpendingDetails, nameof(transactionData.SpendingDetails));
+
+            lock (this.lockObject)
+            {
+                this.outpointLookup.Remove(new OutPoint(transactionData.Id, transactionData.Index));
```

Comment:
Is it a good idea to new up a new `OutPoint` in order to remove it?

Reply:
Is there perhaps a better way to do this?

Reply:
That's the best way.

The `outpointLookup` object is a `Dictionary<OutPoint, TransactionData>` so you can only remove an item by specifying the key, here an `OutPoint`.
The OutPoint class overrides the `==` operator so comparison is serving our purpose here.
https://github.com/stratisproject/StratisBitcoinFullNode/blob/13b272602e63c33fee51ed0f5dc4dc29646a24b0/src/NBitcoin/Transaction.cs#L129-L140

Reply:
Yes the implicit operator will work but if you want to remove the item without a new instance of `Outpoint` you can do this:

```                
var toRemove = this.outpointLookup.FirstOrDefault(op => op.Key.Hash == transactionData.Id && op.Key.N == transactionData.Index);
this.outpointLookup.Remove(toRemove.Key);
```

Reply:
true, but isn't `var toRemove` a new instance of `OutPoint`?

Reply:
In this case `toRemove` will be the actual intance in the dictionary that we want to remove.